sch_yang_condition.c

```diff
@@ -0,0 +1,923 @@
+            *presult = result;
+            result.value = NULL;
+        }
+        else if (g_strcmp0 (xnode->name, "boolean") == 0)
+        {
+            sch_yang_process_arg_list (instance, root, path, step_path, xnode, &result,
+                                       depth, flags);
+            presult->result = result.result;
+        }
+        break;
+    case XPATH_TYPE_STEP:
+        if (xnode->left)
+        {
+            new_step_path = g_strdup (step_path ? : path);
+            sch_yang_process_xnode (instance, root, path, new_step_path, xnode->left,
+                                    &result, depth + 1, flags);
+            presult->result = result.result;
+            if (result.result)
+            {
+                presult->value = result.value;
+                result.value = NULL;
+                if (presult->value)
+                {
+                    g_free (new_step_path);
+                    new_step_path = g_strdup (presult->value);
+                }
+            }
+            else
+            {
+                g_free (result.value);
+                result.value = NULL;
+            }
+
+            if (xnode->right)
+            {
+                sch_yang_process_xnode (instance, root, path, new_step_path, xnode->right,
+                                        &result, depth + 1, flags);
+                presult->result = presult->result & result.result;
+
+                g_free (_path);
+                if (result.result)
+                {
+                    g_free (presult->value);
+                    presult->value = result.value;
+                    result.value = NULL;
+                }
+                else
+                {
+                    g_free (result.value);
+                    result.value = NULL;
+                }
+            }
+            g_free (new_step_path);
+        }
+
+        if (depth == 0 && presult->result)
+        {
+            /* This tests if a path exists */
+            sch_yang_step_exists (root, presult->value, &result);
+            presult->result = result.result;
+        }
+        break;
+    case XPATH_TYPE_STRING:
+        presult->value = g_strdup (xnode->string_value);
+        presult->result = true;
+        break;
+    case XPATH_TYPE_NUMBER:
+        presult->value = g_strdup (xnode->number);
+        presult->result = true;
+        break;
+    case XPATH_TYPE_OR:
+    case XPATH_TYPE_AND:
+    case XPATH_TYPE_EQ:
+    case XPATH_TYPE_NE:
+    case XPATH_TYPE_LT:
+    case XPATH_TYPE_LE:
+    case XPATH_TYPE_GT:
+    case XPATH_TYPE_GE:
+    case XPATH_TYPE_PLUS:
+    case XPATH_TYPE_MINUS:
+    case XPATH_TYPE_MULTIPLY:
+    case XPATH_TYPE_DIVIDE:
+    case XPATH_TYPE_MODULO:
+    case XPATH_TYPE_UNARY_MINUS:
+        sch_yang_process_operator (instance, root, path, step_path, xnode, &result,
+                                   depth, flags);
+        presult->result = result.result;
+        break;
+    case XPATH_TYPE_UNION:
```

Comment:
comment as to why these do nothing might be helpful - is it because there is nothing to do or because we don't support them or something else?

Reply:
While "when" and "must" YANG conditions use an XPATH like syntax, they by no means use all the possible XPATH 1.0 key words. I left the unused parse types in case in the future we hit a more complicate "when" or "must" condition.